### ISSUE TITLE
 [BUG] [Version ^2.0] [events.js] Object doesn't support this action.

### ISSUE THREAD

**Issue by grierson** [kevinchappell/formBuilder#431]
Tried using version 2.1.2 and got this error.
```
File -> events.js
Line (.js) -> events.formSaved = new Event('formSaved');
Line (.min.js) -> r.loaded=new Event("loaded")
Error -> JavaScript runtime error: Object doesn't support this action
```
Tried version 2.0 and [example project in the docs](http://formbuilder.readthedocs.io/en/latest/) and got the same error.

Internet browser: IE11

**Comment by etyp** [comment]
Experiencing the same issue. Any update on this?

**Comment by Zlatkovsky** [comment]
Same issue here as well, can't run formBuilder 2.X on IE 11 (whereas the 1.X version does work)

**Comment by ByteMotion** [comment]
I ran into the same problem as described here today on IE 11, IE 10 and IE9. I have tried to resolve the problem by adding an Event polyfill as described here:

https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent

And changing the lines below in the form-builder JS:

events.loaded = new Event('loaded');
events.viewData = new Event('viewData');
events.userDeclined = new Event('userDeclined');
events.modalClosed = new Event('modalClosed');
events.modalOpened = new Event('modalOpened');
events.formSaved = new Event('formSaved');
events.fieldAdded = new Event('fieldAdded');
events.fieldRemoved = new Event('fieldRemoved');
events.fieldRendered = new Event('fieldRendered');

To the following:

events.loaded = new CustomEvent('loaded');
events.viewData = new CustomEvent('viewData');
events.userDeclined = new CustomEvent('userDeclined');
events.modalClosed = new CustomEvent('modalClosed');
events.modalOpened = new CustomEvent('modalOpened');
events.formSaved = new CustomEvent('formSaved');
events.fieldAdded = new CustomEvent('fieldAdded');
events.fieldRemoved = new CustomEvent('fieldRemoved');
events.fieldRendered = new CustomEvent('fieldRendered');

This seemed to have solved the "Object doesn't support this action" but led to another error in IE:
"Promise is undefined"

So I tried polyfilling that problem as well using the following:

https://github.com/stefanpenner/es6-promise

Which removed all errors in IE but nothing happens on screen and unfortunately the formbuilder does not seem to "start".

I really hope that Kevin finds some time to look into this and fix the problem for us. This formBuilder is by far the most configurable and functional from the 10+ jQuery plugins I have tried and looked at.
